src/c++/perf_analyzer/client_backend/openai/openai_client_backend.h

```diff
@@ -0,0 +1,112 @@
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <string>
+
+#include "../../perf_utils.h"
+#include "../client_backend.h"
+#include "openai_client.h"
+#include "openai_infer_input.h"
+
+#define RETURN_IF_TRITON_ERROR(S)       \
+  do {                                  \
+    const tc::Error& status__ = (S);    \
+    if (!status__.IsOk()) {             \
+      return Error(status__.Message()); \
+    }                                   \
+  } while (false)
+
+namespace tc = triton::client;
+namespace cb = triton::perfanalyzer::clientbackend;
+
+namespace triton { namespace perfanalyzer { namespace clientbackend {
+namespace openai {
+
+
+//==============================================================================
+/// OpenAiClientBackend is used to generate load on the serving instance,
+/// which supports OpenAI Chat Completions API
+///
+class OpenAiClientBackend : public ClientBackend {
+ public:
+  /// Create an OpenAI client backend which can be used to interact with the
+  /// server.
+  /// \param url The inference server url and port.
+  /// \param endpoint The endpoint on the inference server to send requests to
+  /// \param protocol The protocol type used.
+  /// \param http_headers Map of HTTP headers. The map key/value indicates
+  /// the header name/value.
+  /// \param verbose Enables the verbose mode.
+  /// \param client_backend Returns a new OpenAiClientBackend
+  /// object.
+  /// \return Error object indicating success or failure.
+  static Error Create(
+      const std::string& url, const std::string& endpoint,
+      const ProtocolType protocol, std::shared_ptr<Headers> http_headers,
+      const bool verbose, std::unique_ptr<ClientBackend>* client_backend);
+
+  /// See ClientBackend::AsyncInfer()
+  Error AsyncInfer(
+      OnCompleteFn callback, const InferOptions& options,
+      const std::vector<InferInput*>& inputs,
+      const std::vector<const InferRequestedOutput*>& outputs) override;
+
+  /// See ClientBackend::ClientInferStat()
+  Error ClientInferStat(InferStat* infer_stat) override;
+
+ private:
+  OpenAiClientBackend(std::shared_ptr<Headers> http_headers)
+      : ClientBackend(BackendKind::OPENAI), http_headers_(http_headers)
+  {
+  }
+
+  void ParseInferStat(
```

Comment:
Unused?

Reply:
I have a follow-on ticket to test and verify client stats. I'm already seeing some issues around our timers. I'm going to add this comment to the ticket

Reply:
This is a function declaration that doesn't have a corresponding implementation. I think it should be removed or implemented.

Reply:
Agreed! Removed.